src/appleseed/foundation/meta/tests/test_imagefilewriter.cpp

```diff
@@ -0,0 +1,224 @@
+            {
+                for (size_t x = 0; x < s; ++x)
+                {
+                    Color4b c;
+                    image->get_pixel(x, y, c);
+                    EXPECT_EQ(fill_color, c);
+                }
+            }
+        }
+    }
+    
+    TEST_CASE(WriteEXR_CorrectlyWritesImagePixels)
+    {
+        const std::string image_file_path = "unit tests/outputs/test_imagefilewriter.exr";
+        const size_t s = 16;
+        const Color4b fill_color(50, 100, 150, 42);
+                        
+        {
+            write_image<Color4b>(fill_color, PixelFormatUInt8, s, image_file_path);
+        }
+                
+        {
+            GenericImageFileReader reader;
+            std::unique_ptr<Image> image(reader.read(image_file_path.c_str()));
+                    
+            for (size_t y = 0; y < s; ++y)
+            {
+                for (size_t x = 0; x < s; ++x)
+                {
+                    Color4b c;
+                    image->get_pixel(x, y, c);
+                    EXPECT_EQ(fill_color, c);
+                }
+            }
+        }
+    }
+    
+    TEST_CASE(WriteHDR_CorrectlyWritesImagePixels)
+    {
+        const std::string image_file_path = "unit tests/outputs/test_imagefilewriter.hdr";
+        const size_t s = 16;
+        const Color3b fill_color(50, 100, 150);
+                
+        {
+            write_image<Color3b>(fill_color, PixelFormatUInt16, s, image_file_path);
+        }
+        
+        {
+            GenericImageFileReader reader;
+            std::unique_ptr<Image> image(reader.read(image_file_path.c_str()));
+            
+            for (size_t y = 0; y < s; ++y)
+            {
+                for (size_t x = 0; x < s; ++x)
+                {
+                    Color3b c;
+                    image->get_pixel(x, y, c);
+                    EXPECT_EQ(fill_color, c);
+                }
+            }
+        }
+    }
+    
+    TEST_CASE(WriteJPG_ApproximatelyWritesImagePixels)
+    {
+        const std::string image_file_path = "unit tests/outputs/test_imagefilewriter.jpg";
+        const size_t s = 16;
+        const Color3b fill_color(50, 100, 150);
+                
+        {
+            write_image<Color3b>(fill_color, PixelFormatUInt8, s, image_file_path);
+        }
+        
+        {
+            GenericImageFileReader reader;
+            std::unique_ptr<Image> image(reader.read(image_file_path.c_str()));
+            
+            for (size_t y = 0; y < s; ++y)
+            {
+                for (size_t x = 0; x < s; ++x)
+                {
+                    Color3b c;
+                    image->get_pixel(x, y, c);
+                    for (size_t i = 0; i < 3; ++i)
+                    {
+                        EXPECT_TRUE((c[i]-fill_color[i]) < 2 || (fill_color[i] - c[i]) < 2);
+                    }
+                }
+            }
+        }
+    }
+    
+    // Used Color3b instead of Color4b; Color4b does not work. See PR #2779
```

Comment:
I suggest you remove this command and create an issue on Github for this

Reply:
Sure, I will do that!